### PR TITLE
[5.6] Coding style cleanups inside auth traits

### PR DESCRIPTION
**Pull request by neochief**
During the migration from 5.4 I noticed few minor issues inside auth-related traits. I hope the changes are self-explanatory.

### CODE REVIEW

File: src/Illuminate/Foundation/Auth/ThrottlesLogins.php [hunk 1]
```diff
@@ -51,7 +50,7 @@ protected function sendLockoutResponse(Request $request)
         );
 
         throw ValidationException::withMessages([
-            $this->username() => [Lang::get('auth.throttle', ['seconds' => $seconds])],
+            $this->username() => [trans('auth.throttle', ['seconds' => $seconds])],
```
Comment (mfn) [inline src/Illuminate/Foundation/Auth/ThrottlesLogins.php]: IMHO: the less helper in the framework code, the better

Comment (neochief) [reply src/Illuminate/Foundation/Auth/ThrottlesLogins.php]: I've made a search across all framework and there were no other usages of Lang facade. There are several usages of trans() helper, though.

**Comment by taylorotwell** [comment]
Changed it back to a facade.

**Comment by neochief** [comment]
So, is the facade more preferable than helper? Because that second file does use `trans()`: https://github.com/laravel/framework/blob/5.6/src/Illuminate/Foundation/Auth/AuthenticatesUsers.php#L133

**Comment by sisve** [comment]
Isn't changing sendFailedLoginResponse to return void a breaking change? I mean, what should everyone that has an existing sendFailedLoginResponse that returns a Response do? Are they no longer writing valid code?

**Comment by neochief** [comment]
Well, the function itself wasn't changed. I doubt that any code will actually break because of the change.

**Comment by sisve** [comment]
That is not how it works. The api documentation will now state that the method is supposed to return void. This is not the case for how it was before this case.

You've changed the documented return values for a method, thus everyone that has overridden the method needs to adapt to this new signature and modify their code accordingly. It's a breaking change if they have to change their code.

How does this change work with the login method that has a case where it returns whatever sendFailedLoginResponse returns? Should this now be removed since we've changed sendFailedLoginResponse to a void method? What if someone changes it tomorrow based solely on the phpdocs we just wrote? Then everyone that has overridden the method and actually returned a response will have to rewrite their logic to use exceptions (for non-exceptional logic) and catch this exception in the error handler, and there return the response they want. Once again, they have to rewrite their code to match the logic.

**Comment by neochief** [comment]
Thanks for explaining the problem in details. I don't disagree, but I also don't really know how you guys address this type of things. For me, it seems like a bug in the phpdoc. According to the contribution guide, one should PR minor fixes into latest stable branches, which I did.

**Comment by laurencei** [comment]
@sisve 

> The api documentation will now state that the method is supposed to return void. This is not the case for how it was before this case.

But it *was* returning void?

> It's a breaking change if they have to change their code.

I cant think of a situation where this would result in anyone having to change code?

At what stage is a "bug fix" a breaking change? technically any single change is a change in behavior - so you've got to draw the line somewhere...?

**Comment by sisve** [comment]
@laurencei The phpdocs is part of the method signature and tells what the method _can_ do. It applies to everyone that has overridden the method too. This method never returned void, it always threw an exception and counted on the exception handler to generate the response.

Anyone could have overridden the sendFailedLoginResponse in their controller to return any response they wanted. While this technically still works _for now_, what happens tomorrow when someone removes the return in `return $this->sendFailedLoginResponse($request);` in the login method because it's documented to return void? Everyone that has overridden sendFailedLoginResponse and returns a response will no longer see their custom response.

These changes added a couple of `@throws` which could be helpful for documentation, but since php is sane (I rarely say that...) and doesn't have checked exceptions, and allow anyone to throw anything anyway, it doesn't matter that much. It changes a facade call to a helper call. It wrongly changes a return type. There's nothing in this that is classified as a bug fix, so the argumentation how to handle bug fixes isn't relevant here.

**Comment by neochief** [comment]
Just FYI, there are two reasons reason why I had put a void return type there:
1. Because it does return void.
2. To keep it consitent with a similar method inside the ThrottlesLogins class: https://github.com/laravel/framework/blob/5.6/src/Illuminate/Foundation/Auth/ThrottlesLogins.php#L44

**Comment by sisve** [comment]
1. No, it does not return void. It throws an exception. Those are two different things.
2. I understand the consistency argument, but https://github.com/laravel/framework/pull/20567 is an equally breaking change. We failed to catch that change from `@return \Illuminate\Http\RedirectResponse` to `@return void` which has the exact same problem as the change in this issue; people that previously had overridden the sendLockoutResponse method would need to rewrite their logic to go via the exception handler instead of just returning the response.

**Comment by a-komarev** [comment]
@sisve maybe I'm missing something, but you are contradicting yourself. Return type and throws are different things. Current method implementation doesn't have any return statement, so it's returns void. What it throws is completely another thing and described in DocBlock `@throws` statement and isn't related to `@return` statement in any way.

**Comment by sisve** [comment]
Phpdocs doesn't describe what the method is currently doing, but what it is allowed to do. It tells about expectations of the method. These expectations are also applied to everyone that has overridden the method. For example, you're not allowed to make up new parameters, and a toArray() method is not allowed to start returning an integer. In this case, the method is expected to return a Response object.

The current implementation does a somewhat ugly thing, it throws an exception and let's the exception handler transform it into a response. That doesn't mean that users that has already overridden the method has chosen to apply the same logic, to go via the exception handler. They could just as well have returned a Response instance.

Look at the login method with this current implementation. At the end we're doing a `return $this->sendFailedLoginResponse($request);`. This is clearly wrong if sendFailedLoginResponse is a void method, and someone could PR that into just being `$this->sendFailedLoginResponse($request);` without it being a change of logic (since the docs claim that sendFailedLoginResponse is a void method). What happens to everyone that returns a Response object in sendFailedLoginResponse then? It would stop working. And what does should the `@return` declaration of the login method be if the sendFailedLoginResponse  was an actual void method?

> Current method implementation doesn't have any return statement, so it's returns void. 

This is a faulty assumption. The method signatures are not tightly bound to the actual implementations. Interfaces and abstract methods are an obvious example for this. Another would be code that are build to be extendable, which almost everything in the framework is.

    class RNG {
        /**
         * @return int
         */
        public function getRandomNumber() {
            throw new \Exception('Someone should probably implement this method');
        }
    }
    
    class TestRNG extends RNG {
        public function getRandomNumber() {
            return 4;
        }
    }

We cannot change RNG::getRandomNumber() to return void, because there is code that extends it and actually returns the expected value. Everyone that calls $rng->getRandomNumber() expects a number to be returned (or an exception to be thrown). Changing the method to a void method means that everyone that overrides it, and everyone that uses the return value, no longer conforms with the documented behavior of the method.

Thus, the method should not be changed to void. And thus, the sendFailedLoginResponse method this PR modified should not be changed to void either.

**Pull request merged**